beaker/src/depfile_generator.rs

```diff
@@ -0,0 +1,285 @@
+    let mut prerequisites = Vec::new();
+
+    // Add input files
+    for input_file in input_files {
+        prerequisites.push(input_file.clone());
+    }
+
+    // Add stamp files
+    for stamp_path in stamp_info.all_stamps() {
+        prerequisites.push(stamp_path.to_path_buf());
+    }
+
+    // Generate depfile content
+    let content = generate_depfile_content(targets, &prerequisites)?;
+
+    // Write depfile atomically
+    write_depfile_atomically(depfile_path, &content)?;
+
+    Ok(())
+}
+
+/// Generate the content string for a depfile
+fn generate_depfile_content(targets: &[PathBuf], prerequisites: &[PathBuf]) -> Result<String> {
+    if targets.is_empty() {
+        return Ok(String::new());
+    }
+
+    // Format targets
+    let target_strs: Vec<String> = targets.iter().map(|p| escape_path_for_make(p)).collect();
+    let targets_line = target_strs.join(" ");
+
+    // Format prerequisites
+    let prereq_strs: Vec<String> = prerequisites
+        .iter()
+        .map(|p| escape_path_for_make(p))
+        .collect();
+    let prereqs_line = prereq_strs.join(" ");
+
+    // Generate Make rule
+    let content = if prereqs_line.is_empty() {
+        format!("{targets_line}:\n")
+    } else {
+        format!("{targets_line}: {prereqs_line}\n")
+    };
+
+    Ok(content)
+}
+
+/// Escape a path for use in Makefiles
+///
+/// Make requires spaces and some special characters to be escaped
+fn escape_path_for_make(path: &Path) -> String {
+    let path_str = path.to_string_lossy();
+
+    // Escape spaces, dollar signs, and backslashes
+    path_str
+        .replace('\\', "\\\\") // Escape backslashes first
+        .replace(' ', "\\ ") // Escape spaces
+        .replace('$', "$$") // Escape dollar signs
+}
+
+/// Write depfile content atomically using temp file + rename
+fn write_depfile_atomically(depfile_path: &Path, content: &String) -> Result<()> {
+    // Create parent directory if needed
+    if let Some(parent) = depfile_path.parent() {
+        fs::create_dir_all(parent)?;
+    }
+
+    // Write to temporary file first
+    let temp_path = depfile_path.with_extension("tmp");
+    fs::write(&temp_path, content)?;
+
+    // Atomic rename
+    fs::rename(temp_path, depfile_path)?;
+
+    Ok(())
+}
+
+/// Get the output files that will be created for detection
+pub fn get_detection_output_files(
+    input_path: &Path,
+    crop_classes: &std::collections::HashSet<crate::config::DetectionClass>,
+    bounding_box: bool,
+    metadata: bool,
+    output_dir: Option<&str>,
+) -> Vec<PathBuf> {
+    let mut outputs = Vec::new();
+
+    let input_stem = input_path
+        .file_stem()
+        .and_then(|s| s.to_str())
+        .unwrap_or("output");
+
+    let base_dir = if let Some(output_dir) = output_dir {
+        PathBuf::from(output_dir)
+    } else {
+        input_path.parent().unwrap_or(Path::new(".")).to_path_buf()
+    };
+
+    // Add crop outputs - detection uses "crop" suffix regardless of class
```

Comment:
Is that right? I thought it was input_bird_crop for example. Otherwise do they clash?
Maybe it's a bug